From 37ab5ade718741f052b6b8210ff452d8cb04e5d9 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Sat, 19 Mar 2016 09:04:34 +0100
Subject: [PATCH 279/538] Revert "ARM: mm: fix support for HW coherent systems
 in PL310 cache"

This patch reverts the PL310 removal of L2 operations due to HWCC
system. This is done because of a new (and dymanic) way to find this
configuration and remove the L2 operations.
The new implementation is introduced in the followin patches.

This reverts 'commit c4e4d88303a0 ("ARM: dma-mapping: Don't use
outer_flush_range when the L2C is coherent")'.

Change-Id: I4bd2c87396971a7f0969441f2d7c1b42d010bb09
Signed-off-by: Lior Amsalem <alior@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28513
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
---
 arch/arm/mm/cache-l2x0.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mm/cache-l2x0.c b/arch/arm/mm/cache-l2x0.c
index 964a09a..493692d 100644
--- a/arch/arm/mm/cache-l2x0.c
+++ b/arch/arm/mm/cache-l2x0.c
@@ -1271,9 +1271,9 @@ static const struct l2c_init_data of_l2c310_data __initconst = {
 };
 
 /*
- * This is a variant of the of_l2c310_data with .sync and .flush_range set to
- * NULL. Outer sync and flush range operations are not needed when the system
- * is I/O coherent, and potentially harmful in certain situations (PCIe/PL310
+ * This is a variant of the of_l2c310_data with .sync set to
+ * NULL. Outer sync operations are not needed when the system is I/O
+ * coherent, and potentially harmful in certain situations (PCIe/PL310
  * deadlock on Armada 375/38x due to hardware I/O coherency). The
  * other operations are kept because they are infrequent (therefore do
  * not cause the deadlock in practice) and needed for secondary CPU
@@ -1292,6 +1292,7 @@ static const struct l2c_init_data of_l2c310_coherent_data __initconst = {
 	.outer_cache = {
 		.inv_range   = l2c210_inv_range,
 		.clean_range = l2c210_clean_range,
+		.flush_range = l2c210_flush_range,
 		.flush_all   = l2c210_flush_all,
 		.disable     = l2c310_disable,
 		.resume      = l2c310_resume,
-- 
1.9.1

